ndb_api/auth.go

```diff
@@ -0,0 +1,67 @@
+/*
+Copyright 2022-2023 Nutanix, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ndb_api
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/nutanix-cloud-native/ndb-operator/ndb_client"
+	ctrllog "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+// Validates the auth credentials against the 'auth/validate' endpoint
+// Returns the response along with an error (if any)
+func AuthValidate(ctx context.Context, ndbClient *ndb_client.NDBClient) (authValidateResponse AuthValidateResponse, err error) {
+	log := ctrllog.FromContext(ctx)
+	log.Info("Entered ndb_api.AuthValidate")
+	if ndbClient == nil {
+		err = errors.New("nil reference: received nil reference for ndbClient")
+		log.Error(err, "Received nil ndbClient reference")
+		return
+	}
+	res, err := ndbClient.Get("auth/validate")
+	if err != nil || res == nil || res.StatusCode != http.StatusOK {
+		if err == nil {
```

Comment:
Do you need to conditionally log the res, err, status etc? We can just dump all three [res, err, status] if either of (err != nil || res == nil || res.StatusCode != http.StatusOK) is true. Would make the code simpler.

Reply:
These are specific errors that are being returned to the caller too, I'm not conditionally logging them. In order to provide the exact type of error to the caller, this if/else case might be necessary. WDYT?